stack/provider_requirements.go

```diff
@@ -0,0 +1,6 @@
+package stack
+
+type ProviderRequirement struct {
+	Source             string
+	VersionConstraints []string
```

Comment:
I think it would be helpful to parse this into a `version.Constraints`

Reply:
We'll address that when we start to use it when implementing references